DataDog/datadog-operator
# feat: support datadog profiler in operator

### What does this PR do?

Add Datadog profiling support in the Datadog Operator binary.

Set `-profiling-enabled=true` to enable Datadog profiling in the Operator binary.

### Motivation

Ease Operator performance investigation

### Additional Notes

The PR refactor the `main()` and introduce `func run(...) err` function to allow using `defer func()`.

### Describe your test plan

Deploy the operator with the option `-profiling-enabled=true`, then deploy the Agent with APM/profiling enable thanks to the `DatadogAgent`.
This setup is a bit tricky because you need configure the "APM service name" before it was created.

So maybe the simpler solution is to:
1. deploy the Operator
2. create a DatadogMetrics to deploy the agent with APM/Profiling enable
3. update the Operator configuration to set   `-profiling-enabled=true` and set the envvars `DD_AGENT_HOST=<service-name>
4. check in Datadog app if you see the profile
